com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/zcl/ZclCluster.java

```diff
@@ -1017,12 +1043,6 @@ public Boolean call() throws Exception {
      * @param listener the {@link ZclAttributeListener} to add
      */
     public void addAttributeListener(ZclAttributeListener listener) {
-        // Don't add more than once.
-        if (attributeListeners.contains(listener)) {
```

Comment:
If I recall this correctly: you have just introduced that check to avoid duplicates. What is the reason to remove this check again? Anyway its a `CopyOnWriteArraySet` so I wonder how duplicates can happen at all here.

Reply:
I think you are getting muddled with another PR recently for a different listener - possibly in the command notification class.

Here ```attributeListeners``` is a ```Set``` so it should not be required to check for duplicates right?